SU2_CFD/include/variables/CVariable.hpp

```diff
@@ -47,2469 +47,2615 @@
    */
-  inline su2double GetSolution_Old(unsigned short val_var) {return Solution_Old[val_var]; }
+  inline su2double GetSolution_Old(Idx_t iPoint, Idx_t iVar) const { return Solution_Old(iPoint,iVar); }
 
   /*!
-   * \brief Get the old solution of the discrete adjoint problem (for multiphysics subiterations=
-   * \param[in] val_var - Index of the variable.
+   * \brief Get the old solution of the discrete adjoint problem (for multiphysics subiterations)
+   * \param[in] iPoint - Point index.
+   * \param[in] iVar - Index of the variable.
    * \return Pointer to the old solution vector.
    */
-  inline su2double GetSolution_Old_Adj(unsigned short val_var) {return Solution_Adj_Old[val_var]; }
+  inline su2double GetSolution_Old_Adj(Idx_t iPoint, Idx_t iVar) const { return Solution_Adj_Old(iPoint,iVar); }
 
   /*!
    * \brief Set the value of the old solution.
-   * \param[in] val_solution_old - Pointer to the residual vector.
+   * \param[in] iPoint - Point index.
+   * \param[in] solution_old - Pointer to the residual vector.
    */
-  inline void SetSolution_Old(su2double *val_solution_old) {
-    for (unsigned short iVar = 0; iVar < nVar; iVar++)
-      Solution_Old[iVar] = val_solution_old[iVar];
+  inline void SetSolution_Old(Idx_t iPoint, const su2double *solution_old) {
+    for (Idx_t iVar = 0; iVar < nVar; iVar++)
+      Solution_Old(iPoint,iVar) = solution_old[iVar];
   }
 
   /*!
    * \overload
-   * \param[in] val_var - Index of the variable.
-   * \param[in] val_solution_old - Value of the old solution for the index <i>val_var</i>.
+   * \param[in] iVar - Index of the variable.
+   * \param[in] iPoint - Point index.
+   * \param[in] solution_old - Value of the old solution for the index <i>iVar</i>.
    */
-  inline void SetSolution_Old(unsigned short val_var, su2double val_solution_old) {Solution_Old[val_var] = val_solution_old; }
+  inline void SetSolution_Old(Idx_t iPoint, Idx_t iVar, su2double solution_old) { Solution_Old(iPoint,iVar) = solution_old; }
 
   /*!
    * \brief Set old variables to the value of the current variables.
    */
-  inline void Set_OldSolution(void) {
-    for (unsigned short iVar = 0; iVar < nVar; iVar++)
-      Solution_Old[iVar] = Solution[iVar];
-  }
+  void Set_OldSolution();
 
   /*!
    * \brief Set variables to the value of the old variables.
    */
-  inline void Set_Solution(void) {
-    for (unsigned short iVar = 0; iVar < nVar; iVar++)
-     Solution[iVar] = Solution_Old[iVar];
-  }
+  void Set_Solution();
 
-  /*!
-   * \brief Set old discrete adjoint variables to the current value of the adjoint variables.
-   */
-  inline void Set_OldSolution_Adj(void) {
-    for (unsigned short iVar = 0; iVar < nVar; iVar++)
-      Solution_Adj_Old[iVar] = Solution[iVar];
-  }
+//  /*!
+//   * \brief Set old discrete adjoint variables to the current value of the adjoint variables.
+//   * \param[in] iPoint - Point index.
+//   */
+//  inline void Set_OldSolution_Adj(Idx_t iPoint) {
```

Comment:
here

Reply:
I would say we remove whatever is not used at the moment.

Reply:
I share that opinion. @economon what do you think?

Reply:
Please don't xD that would make my optimizations more intricate as I would need separate meshes for the direct problem and then merge them for the adjoint... Let's do it all at once, one PR -20k lines it'll be more impressive that way.